Pixel Game/Assets/Scripts/ObjectPool.cs

```diff
@@ -29,26 +15,29 @@ public ObjectPool(GameObject templateObject, int size)
 			var obj = GameObject.Instantiate(templateObject);
 			obj.SetActive(false);
             obj.hideFlags = HideFlags.HideInHierarchy;
-            GameObject.DontDestroyOnLoad(obj);
+            //GameObject.DontDestroyOnLoad(obj);
 			pool.Add(obj);
 		}
 	}
 
-	public GameObject Spawn(Vector3 pos)
+	public GameObject Spawn(Vector2 pos)
 	{
 		//Find an inactive object
 		foreach (var obj in pool)
 		{
 			if (!obj.activeInHierarchy)
 			{
-				obj.transform.position = pos;
+				//Z pos needs to be -1 for glitch-free animation (Why? Can we fix this?)
```

Comment:
Most likely the -1 z-position is needed since this object pool is using the particle system right?

The particles start at z-pos 0 and shoot towards a positive z-value which leads to the particles not being visible since the camera is at -10 z-pos looking at the scene on 0 z-pos and cant detect positive z objects. 

The Fix to this is to remake the particle effect so it starts at z-pos 0 and shoots out towards a **negative** z-value.

Reply:
Interesting, good to know. Yes it's the particle system. I did some testing just now and it seems this is not a problem anymore. Setting z to 0 works fine. Maybe some camera settings changed?

Reply:
I will fix the issue so it does not appear somehow in the future